pytr/transactions.py

```diff
@@ -1,60 +1,174 @@
+    """ The language for the CSV header / JSON keys. """
+
+    date_with_time: bool = True
+    """ Include the timestamp in ISO8601 format in the date field. """
+
+    decimal_localization: bool = False
+    """ Whether to localize the decimal format. If enabled, decimal fields will be string values. """
+
+    csv_delimiter: str = ";"
+
+    def __post_init__(self):
+        self._log = get_logger(__name__)
+
+        if self.lang == "auto":
+            locale = getdefaultlocale()[0]
+            if locale is None:
+                self.lang = "en"
+            else:
+                self.lang = locale.split("_")[0]
+
+        if self.lang not in SUPPORTED_LANGUAGES:
+            self._log.info(f'Language not yet supported "{self.lang}", defaulting to "en"')
+            self.lang = "en"
+
+        self._translate = setup_translation(language=self.lang)
+
+    def _decimal_format(self, value: Optional[float], quantization: bool = True) -> Union[str, float, None]:
+        if value is None:
+            return None
+        return (
+            format_decimal(value, locale=self.lang, decimal_quantization=quantization)
+            if self.decimal_localization
+            else value
+        )
+
+    def _localize_keys(self, txn: _SimpleTransaction) -> dict[str, Any]:
+        if self.lang is None:
+            return
+        return {self._translate(value): txn[key] for key, value in CSVCOLUMN_TO_TRANSLATION_KEY.items()}  # type: ignore[literal-required]
+
+    def fields(self) -> list[str]:
+        return [self._translate(value) for key, value in CSVCOLUMN_TO_TRANSLATION_KEY.items()]
+
+    def from_event(self, event: Event) -> Iterable[dict[str, Any]]:
+        """
+        Given an event, produces one or more JSON objects representing a transaction. The returned object contains
+        the given fields, localized in the selected language.
+
+        - `date`
+        - `type`
+        - `value`
+        - `note`
+        - `isin`
+        - `shares`
+        - `fees`
+        - `taxes`
+        """
+
+        if event.event_type is None:
+            return
+
+        if event.event_type == ConditionalEventType.TRADE_INVOICE:
+            assert event.value is not None, event
+            event.event_type = PPEventType.BUY if event.value < 0 else PPEventType.SELL
+
+        kwargs: _SimpleTransaction = {
+            "date": event.date.isoformat() if self.date_with_time else event.date.date().isoformat(),
+            "type": self._translate(event.event_type.value) if isinstance(event.event_type, PPEventType) else None,
+            "value": self._decimal_format(event.value),
+            "note": self._translate(event.note) + " - " + event.title if event.note is not None else event.title,
+            "isin": event.isin,
+            "shares": self._decimal_format(event.shares, False),
+            "fees": self._decimal_format(-event.fees) if event.fees is not None else None,
+            "taxes": self._decimal_format(-event.taxes),
```

Comment:
```suggestion
            "taxes": self._decimal_format(-event.taxes) if event.taxes is not None else None,
```

You need to handle None here.